### ISSUE TITLE
Not all cases are covered

### ISSUE THREAD

**Issue by dragos-rosca** [osstotalsoft/nodebb#11]
https://github.com/osstotalsoft/nodebb/blob/bccd134112e7ff2c49822d07ddeefd09d266ad37/packages/knex-filters/src/queryBuilder.js#L61

Hello,
Not all cases are covered, it looks like `knex.count(...)` is not triggered on `onSelect` leaving us not having the global filter (tenant id) added in our counts.
Maybe a `default` and handle it like it's a select for now until we catch all the methods.

**Comment by dragos-rosca** [comment]
this is the code that fails:
```javascript
  generateFromAndWhereClause(queryBuilder, { afterId, filters = {}, direction = 0, sortBy = 'FirstName', sortByValue }) {
    const { firstName, lastName, state, clientId, userClient } = filters

    queryBuilder.from('EOUser')

    if (firstName) {
      queryBuilder.whereRaw('UPPER(FirstName) LIKE ?', `%${firstName.toUpperCase()}%`)
    }

    if (lastName) {
      queryBuilder.whereRaw('UPPER(LastName) LIKE ?', `%${lastName.toUpperCase()}%`)
    }

    if (state !== undefined && state !== null) {
      queryBuilder.whereRaw('COALESCE(IsActive, 0) = ?', state)
    }

    const imposedClientId = userClient?.isSelfPartner ? clientId : userClient?.id
    if (imposedClientId) {
      queryBuilder.where('ClientId', imposedClientId)
    }

    if (afterId) {
      queryBuilder.modify(generateSortByPkClause, { sortBy, pk: 'Id', direction, afterId, sortByValue })
    }
  }

// This is the function that fails:
  async getUserListTotalCount(filters = {}) {
    return await this.knex.count('Id', { as: 'TotalCount' }).modify(this.generateFromAndWhereClause, { filters }).first()
// I left in the modify but it fails with 0 filters
// Please see that .first() at the end
  }
```

**Comment by oncicaradupopovici** [comment]
The problem was not the "count" fn, but rather the usage of "first", which apparently uses a separate query builder method, that is now captured by a default branch, that triggers the onSelect hook.

As a side remark, imho, the usage of "count" and "first" in the same query is awkward and maybe redundand.

**Comment by oncicaradupopovici** [comment]
fixed in @totalsoft/knex-filters@0.3.5
